clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2077,6 +2077,22 @@ void Sema::finalizeSYCLDelayedAnalysis(const FunctionDecl *Caller,
   }
 }
 
+bool Sema::isArrayZeroInitialized(VarDecl *VD, bool CheckValueDependent) {
+  assert(getLangOpts().SYCLIsDevice &&
+         "Should only be called during device compilation");
+  if (VD->isInvalidDecl() || !VD->hasInit() || !VD->hasGlobalStorage())
+    return false;
+  APValue *Value = nullptr;
+  if (VD->getInit() && !VD->getInit()->isValueDependent())
+    Value = VD->evaluateValue();
+  // zero-initialized arrays do not have any initialized elements.
+  if (Value && Value->isArray() && Value->hasArrayFiller() &&
+      Value->getArrayInitializedElts() == 0) {
```

Comment:
minor: coding standard recommends not using braces for single statement in the body of an if.
But otherwise looks good.